client-libraries/rust/modality-reflector-config/src/lib.rs

```diff
@@ -0,0 +1,990 @@
+                    .collect(),
+                override_mutator_attributes: value
+                    .override_mutator_attributes
+                    .into_iter()
+                    .map(Into::into)
+                    .collect(),
+            }
+        }
+    }
+    impl From<refined::PluginsIngest> for PluginsIngest {
+        fn from(value: refined::PluginsIngest) -> Self {
+            Self {
+                collectors: value
+                    .collectors
+                    .into_iter()
+                    .map(|(k, v)| (k, v.into()))
+                    .collect(),
+                importers: value
+                    .importers
+                    .into_iter()
+                    .map(|(k, v)| (k, v.into()))
+                    .collect(),
+            }
+        }
+    }
+    impl From<refined::PluginsMutation> for PluginsMutation {
+        fn from(value: refined::PluginsMutation) -> Self {
+            Self {
+                mutators: value
+                    .mutators
+                    .into_iter()
+                    .map(|(k, v)| (k, v.into()))
+                    .collect(),
+            }
+        }
+    }
+    impl From<refined::PluginsIngestMember> for PluginsIngestMember {
+        fn from(value: refined::PluginsIngestMember) -> Self {
+            Self {
+                timeline_attributes: value.timeline_attributes.into(),
+                metadata: value.metadata,
+            }
+        }
+    }
+    impl From<refined::PluginsMutationMember> for PluginsMutationMember {
+        fn from(value: refined::PluginsMutationMember) -> Self {
+            Self {
+                mutator_attributes: value.mutator_attributes.into(),
+                metadata: value.metadata,
+            }
+        }
+    }
+
+    impl From<refined::AvailablePorts> for AvailablePorts {
+        fn from(value: refined::AvailablePorts) -> Self {
+            Self {
+                any_local: value.any_local,
+                ranges: value
+                    .ranges
+                    .into_iter()
+                    .map(|inclusive_range| [inclusive_range.start(), inclusive_range.end()])
+                    .collect(),
+            }
+        }
+    }
+}
+
+/// Public-facing, more-semantically-enriched configuration types
+mod refined {
+    use super::TomlValue;
+    pub use modality_ingest_client::types::AttrVal;
+    use std::collections::BTreeMap;
+    use std::fmt;
+    use std::path::PathBuf;
+    use std::str::FromStr;
+    use std::time::Duration;
+    use url::Url;
+
+    /// A local string-ish AttrKey value.
+    #[derive(Clone, Debug, PartialEq, Eq, PartialOrd)]
+    pub struct AttrKey(String);
+
+    impl AttrKey {
+        fn new(k: String) -> Self {
+            Self(k)
+        }
+    }
+
+    impl std::fmt::Display for AttrKey {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+            write!(f, "{}", self.0)
+        }
+    }
```

Comment:
I added a different `AttrKey` that is string-ish rather than int-ish like it is in the SDK. Is it confusing that they're called the same thing even though they're in different places. Do we already have names to distinguish between string-ish AttrKey and int-ish AttrKey anywhere else?

Reply:
The int-ish `AttrKey` is an implementation detail - a caching technique specific to the `CBOR` ingest protocol.  Re: name differences, crate-and-mod-path differences is distinguishing enough for me, but your mileage may vary. 

If we need a type alias for the CBOR one, then it's probably fine to invent one.

Reply:
Okay. In that case I'm not going to try to string-ish-ify the ingest client `AttrKey` right not since that'd be either a breaking change or take some tricksy doing to not be breaking, I'll just leave this with the two different module paths.

Reply:
Now that I notice you're pulling in the whole ingest crate to avoid duplicating the attribute-value-related types, I can see why this would be a point of confusion.

Reply:
Perhaps a simple and moderately clarifying thing to do would be to move the protocol-specific `modality_ingest_client::types::AttrKey` to be at `modality_ingest_client::protocol::AttrKey` instead.

Reply:
You're also making the right choice of not string-ifying the ingest client's types further at this time, I think.